ui/js/util/redux-actions.js

```diff
@@ -0,0 +1,23 @@
+export const handleActions = (actionMap, defaultState) => {
```

Comment:
What do you think of allowing this wrapper to handle the `Object.assign` that every reducer does? Something like...
```
export const handleActions = (actionMap, defaultState) => {
  return (state = defaultState, action) => {
    const handler = actionMap[action.type];
    const newState = handler ? handler(state, action) : {}
    return Object.assign({}, state, newState);
  };
};
```

In writing this, I noticed that the reducers you added in `shape_shift.js` do not include the `Object.assign` calls. Aren't those always necessary?

Reply:
Not if using the spread operator `...`.
https://twitter.com/dan_abramov/status/688087202312491008

Reply:
We could still add that to the `handleActions` function anyway. So we don't have to add `...state` in every reducer handler

Reply:
`Object.assign` isn't a deep clone either. `Object.assign({}, state, { foo: "bar"})` and `{ ...state, { foo: "bar" }` are functionally equivalent. 

(Correct me if above seems wrong to you.)

Reply:
Yeah they are the same. I believe babel transpiles the spread operator into object.assign